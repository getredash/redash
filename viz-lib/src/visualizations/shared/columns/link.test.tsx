import React from "react";
import enzyme from "enzyme";

import Column from "./link";

function findByTestID(wrapper: any, testId: any) {
  return wrapper.find(`[data-test="${testId}"]`);
}

function mount(column: any, done: any) {
  return enzyme.mount(
    <Column.Editor
      // @ts-expect-error ts-migrate(2322) FIXME: Type '{ visualizationName: string; column: any; on... Remove this comment to see the full error message
      visualizationName="Test"
      column={column}
      onChange={changedColumn => {
        expect(changedColumn).toMatchSnapshot();
        done();
      }}
    />
  );
}

describe("Visualizations -> Table -> Columns -> Link", () => {
  describe("Editor", () => {
    test("Changes URL template", done => {
      const el = mount(
        {
          name: "a",
          linkUrlTemplate: "{{ @ }}",
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Link.UrlTemplate")
        .last()
        .find("input")
        .simulate("change", { target: { value: "http://{{ @ }}/index.html" } });
    });

    test("Changes text template", done => {
      const el = mount(
        {
          name: "a",
          linkTextTemplate: "{{ @ }}",
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Link.TextTemplate")
        .last()
        .find("input")
        .simulate("change", { target: { value: "Text of {{ @ }}" } });
    });

    test("Changes title template", done => {
      const el = mount(
        {
          name: "a",
          linkTitleTemplate: "{{ @ }}",
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Link.TitleTemplate")
        .last()
        .find("input")
        .simulate("change", { target: { value: "Title of {{ @ }}" } });
    });

    test("Makes link open in new tab ", done => {
      const el = mount(
        {
          name: "a",
          linkOpenInNewTab: false,
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Link.OpenInNewTab")
        .last()
        .find("input")
        .simulate("change", { target: { checked: true } });
    });
  });
});
