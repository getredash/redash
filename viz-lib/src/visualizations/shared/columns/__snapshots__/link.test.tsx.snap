// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualizations -> Table -> Columns -> Link Editor Changes URL template 1`] = `
Object {
  "linkUrlTemplate": "http://{{ @ }}/index.html",
}
`;

exports[`Visualizations -> Table -> Columns -> Link Editor Changes text template 1`] = `
Object {
  "linkTextTemplate": "Text of {{ @ }}",
}
`;

exports[`Visualizations -> Table -> Columns -> Link Editor Changes title template 1`] = `
Object {
  "linkTitleTemplate": "Title of {{ @ }}",
}
`;

exports[`Visualizations -> Table -> Columns -> Link Editor Makes link open in new tab  1`] = `
Object {
  "linkOpenInNewTab": true,
}
`;
